### DISCUSSION TITLE
Get accent colors from wallpaper?

### DISCUSSION THREAD

**Discussion by arun54321** [GradienceTeam/Gradience#26] [Ideas]
Similar to android 12's material you.

**Comment by daudix** [comment]
Mockup #25

**Comment by 0xMRTT** [comment]
Hi @arun54321 

This feature is fully implemented.

**Reply by RpxdYTX** [reply]
Question: could the changes of something like Variety or Wallch to the curtent background affect the color pallete of the Adwaita preset? Could only parts of the theme be changed by that (like, keeping the dark background style but whenever the accent colors change, the background is changed slightly to conform with the accents)?

**Reply by 0xMRTT** [reply]
You mean that gradience will modify the background ?

**Reply by arun54321** [reply]
I understand that there is option to extract colors/monet from the chosen image now but I don't see option to pick colors from wallpaper.

**Reply by 0xMRTT** [reply]
You mean, adding a color picker and generating a theme from 1 color ?

**Reply by arun54321** [reply]
Under monet tab, by default it points to adwaita-l.jpg which is not the wallpaper I use.
I want the app to extract the colors from the current wallpaper.

**Reply by 0xMRTT** [reply]
Wich version are you using because it's use the background specified in Dconf `org.gnome.desktop.background.picture-uri`

**Reply by arun54321** [reply]
0.2.2 from flatpak from flathub.

**Reply by 0xMRTT** [reply]
Do you have dconf installed ? If yes, could you check the value of the previous key

**Reply by arun54321** [reply]
I see that feature is already included. May be it is not working because I'm using gnome 43.beta?

**Reply by 0xMRTT** [reply]
Not sure because it's working for me. On Fedora 37 and arch Linux with fcgu

**Reply by arun54321** [reply]
this is what I have. I set wallpaper from nautilus- right click, set as background
![image](https://user-images.githubusercontent.com/42943340/188314936-bbedd6d2-ffd6-4d5d-9263-41d5312f101f.png)

**Reply by 0xMRTT** [reply]
You use f36 or f37 ?

**Reply by 0xMRTT** [reply]
And is background a picture or a directory ?

**Reply by RpxdYTX** [reply]
> You mean that gradience will modify the background ?

Nope, Variety and Wallch allows you to automatically change the background in a fixed interval. So could Monet extract those colors from the wallpaper and apply them?
And maybe allow us to control which parts of the interface or how much of the color gets affected (e.g my preset is something like Sweet Dark theme but blue. Using monet, only the accent colors would be affected, and the background would be slightly tweaked to fit the accents)?

> Under monet tab, by default it points to adwaita-l.jpg which is not the wallpaper I use. I want the app to extract the colors from the current wallpaper.

Exactly.

> this is what I have. I set wallpaper from nautilus- right click, set as background ![image](https://user-images.githubusercontent.com/42943340/188314936-bbedd6d2-ffd6-4d5d-9263-41d5312f101f.png)

I use Variety, but idk anything about `org.gnome.desktop.picture-uri` or what does Variety do to this setting whatsoever.

**Comment by 0xMRTT** [comment]
Feel free to join matrix or discord

https://matrix.to/#/#Gradience:matrix.org

https://discord.com/invite/rwNDGPJf
